libs/pageserver_api/src/key.rs

```diff
@@ -967,12 +967,12 @@ mod tests {
         let mut rng = rand::rngs::StdRng::seed_from_u64(42);
 
         let key = Key {
-            field1: rng.gen(),
-            field2: rng.gen(),
-            field3: rng.gen(),
-            field4: rng.gen(),
-            field5: rng.gen(),
-            field6: rng.gen(),
+            field1: rng.r#gen(),
+            field2: rng.r#gen(),
+            field3: rng.r#gen(),
+            field4: rng.r#gen(),
+            field5: rng.r#gen(),
+            field6: rng.r#gen(),
```

Comment:
```suggestion
            field1: rng.random(),
            field2: rng.random(),
            field3: rng.random(),
            field4: rng.random(),
            field5: rng.random(),
            field6: rng.random(),
```

per https://docs.rs/rand/latest/rand/trait.Rng.html#method.gen, there's a new alias `random` for this function to avoid this clash with the 'gen' keyword.

Reply:
This function is only available in `rand` 0.9 which we don't use (yet).